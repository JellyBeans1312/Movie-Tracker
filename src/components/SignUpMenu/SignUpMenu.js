import React, { Component } from 'react';
import { SignUp } from '../SignUp/SignUp'
import { login, showError } from '../../actions';
import { connect } from 'react-redux';
import { getUser, addUser } from '../../api/apiCalls'
import { Redirect } from 'react-router'
import './SignUpMenu.css'
import PropTypes from 'prop-types';

export class SignUpMenu extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      email: '',
      password: '',
      display: '',
      error: null
    };
  }

  handleAddChange = (e) => {
    const {name, value} = e.target
    this.setState({ [name]: value })
  }

  handleAdd = async (e) => {
    try {
      e.preventDefault()
      const response = await addUser(this.state);
      console.log(response)
      if (response === true) {
        this.props.login(this.state)
        this.setState({display: 'loggedIn'})
      } else {
        this.setState({ error: "Email has already been used" });
      }
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  render() {
    const isLoggedIn = this.state.display === 'loggedIn'
    let view;

    if(!isLoggedIn) {
      view = <SignUp name={this.state.name} email={this.state.email} password={this.state.password} handleAddChange={this.handleAddChange} handleAdd={this.handleAdd}/>
    } else {
      view = <Redirect to='/LoggedIn'/>
    }
    return (
      <section className='sign-up-bar' >
        {view}
        {this.state.error && <h2 className='error'>{this.state.error}</h2>}
      </section>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  login: (user) => dispatch(login(user)),
  showError: (error) => dispatch(showError(error))
});

SignUpMenu.propTypes = {
  login: PropTypes.func.isRequired,
  showError: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired
}

export default connect(null, mapDispatchToProps)(SignUpMenu)
