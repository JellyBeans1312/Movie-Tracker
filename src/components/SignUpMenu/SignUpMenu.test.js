import React from 'react';
import { shallow } from 'enzyme';
import { SignUpMenu, mapStateToProps, mapDispatchToProps} from './SignUpMenu'
import { showError, login } from '../../actions'

describe('SignUpMenu', () => {
  let wrapper;
  let mockLogin;
  let mockShowErrorFunc;
  let mockShowError;
  let mockUser;

  beforeEach(() => {
    mockUser = {}
    mockLogin = jest.fn();
    mockShowErrorFunc = jest.fn()
    mockShowError = 'email incorrect';
    wrapper = shallow(<SignUpMenu showError={mockShowErrorFunc} login={mockLogin} showError={mockShowError} user={mockUser}/>);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('SignUpMenu Component', () => {
    it('should update state when handle add change is called', () => {
      const mockEvent = {target: {name: 'Nombre', value:'password'}}
      const expected = 'password';

      wrapper.instance().handleAddChange(mockEvent);

      expect(wrapper.state('Nombre')).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with a login error', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = login(1, 'nombre@nombre.com', 'password')
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.login(1, 'nombre@nombre.com', 'password')

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})