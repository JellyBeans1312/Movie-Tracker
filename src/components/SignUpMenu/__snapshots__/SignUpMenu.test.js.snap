// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpMenu should match snapshot 1`] = `
<section
  className="sign-up-bar"
>
  <SignUp
    email=""
    handleAdd={[Function]}
    handleAddChange={[Function]}
    name=""
    password=""
  />
</section>
`;

exports[`SignUpMenu should match snapshot 2`] = `
<section
  className="sign-up-bar"
>
  <SignUp
    email=""
    handleAdd={[Function]}
    handleAddChange={[Function]}
    name=""
    password=""
  />
</section>
`;
